File: setubunrobot/setubunrobot_test.py
import unittest

import setubunrobot


class TestSetubunRobot(unittest.TestCase):
    def test1(self):
        problem = setubunrobot.Problem(
            members=[10, 20, 30, 40, 50], turns=[(2, 4, 10), (1, 3, 15)]
        )
        want = setubunrobot.Answer(beans=[10, 20, 25, 10, 0])
        got = setubunrobot.solve(problem)
        self.assertEqual(want, got)

    def test2(self):
        problem = setubunrobot.Problem(members=[3, 1, 4], turns=[(1, 1, 2)])
        want = setubunrobot.Answer(beans=[2, 0, 0])
        got = setubunrobot.solve(problem)
        self.assertEqual(want, got)


if __name__ == "__main__":
    unittest.main()

File: setubunrobot/setubunrobot_test.go
package setubunrobot_test

import (
	"testing"

	"github.com/hsmtkk/fuzzy-giggle/setubunrobot"
	"github.com/stretchr/testify/assert"
)

func Test1(t *testing.T) {
	p := setubunrobot.Problem{Members: []uint{10, 20, 30, 40, 50}, Turns: []setubunrobot.Turn{{1, 3, 10}, {0, 2, 15}}}
	want := setubunrobot.Answer{[]uint{10, 20, 25, 10, 0}}
	got := setubunrobot.New().Distribute(p)
	assert.Equal(t, want, got)
}

func Test2(t *testing.T) {
	p := setubunrobot.Problem{Members: []uint{3, 1, 4}, Turns: []setubunrobot.Turn{{0, 0, 2}}}
	want := setubunrobot.Answer{[]uint{2, 0, 0}}
	got := setubunrobot.New().Distribute(p)
	assert.Equal(t, want, got)
}
